### ISSUE TITLE
Make email verification and password tokens one-time use only

### ISSUE THREAD

**Issue by infomiho** [wasp-lang/wasp#1933]
According to https://thecopenhagenbook.com/password-reset and https://security.stackexchange.com/a/244855 it makes sense to make the tokens one-time use only to ensure that bad actors can't intercept the URL and use the token to hijack a user's account.

### Implementation 

Since we have `providerData` which has a flexible JSON, we can add two new fields there for the `email` auth provider: `passwordResetToken` and `emailVerificationToken` which we then check against when trying to perform the action.

If the token is in the DB, accept the action and remove it from the DB. If it's not in the DB, refuse the action.

**Comment by Martinsos** [comment]
We can name those better by prefixing them with `lastValid`: `lastValidPasswordResetToken` and `lastValidEmailVerificationToken`, or at least prefixing them with `valid`, or `current` or `latest`. Actually, what we want to communicate is: this is a single token that is out there and hasn't yet been used. It might be expired, but it hasn't been consumed. So what is the best way to describe it? `lastIssued` may be a good prefix then. Yeah, that sounds quite good. But it sounds like we would expect that field to be set even after it was consumed, which is not correct. Ok, so not a good name. Then maybe `outstanding`? Yeah, that sounds quite good. It indicates it is out there, unconsumed. Oh yeah, I think "outstanding" is perfect. It is also better than `lastOutstanding`, because `last` sounds like there can be multiple outstanding ones, but there can't there can be only one outstanding one -> because the older ones become invalid at the moment new one is produced.

TLDR: I would prefix these variables with `outstanding`. I left the reasoning above in case it might be interesting.
